### PR TITLE
Addressing issue #72

### PR DESCRIPTION
**Pull request by ClaudiaComito**
- Modified operations.__reduce_op to take care of axis sanitation and calculation of the "partial" tensor.
- Modified operations.max() and operations.min() accordingly
- Introduced draft of operations.sum()

### CODE REVIEW

File: heat/core/operations.py [hunk 1]
```diff
@@ -383,7 +378,7 @@ def __local_operation(operation, x, out):
     operation(casted.repeat(multiples) if needs_repetition else casted, out=out._tensor__array)
     return out
 
-def __reduce_op(x,partial, op, axis):
+def __reduce_op(x,partial_op, op, axis):
```
Comment (Markus-Goetz) [inline heat/core/operations.py]: Please add a space before partial_op to be PEP-8 compliant

File: heat/core/operations.py [hunk 2]
```diff
@@ -321,6 +309,13 @@ def sqrt(x, out=None):
     """
     return __local_operation(torch.sqrt, x, out)
 
+def sum(x, axis=None):
+    # TODO: document me
```
Comment (Markus-Goetz) [inline heat/core/operations.py]: Would you be so kind to provide the doc-string for the function.
The test TODO can be removed as they are provided by Kai already.

File: heat/core/operations.py [hunk 3]
```diff
@@ -321,6 +309,13 @@ def sqrt(x, out=None):
...
+    # TODO: test me
+    # TODO: make me more numpy API complete
+    # TODO: implement type promotion
```
Comment (Markus-Goetz) [inline heat/core/operations.py]: Type promotion has to be generically handled in __reduce_op, TODO can be moved to said function

File: heat/core/operations.py [hunk 4]
```diff
@@ -231,15 +232,8 @@ def max(x, axis=None):
...
-    #perform sanitation:
-    axis = stride_tricks.sanitize_axis(x.shape,axis)
-    
-    if axis is not None:        
-        max_axis = x._tensor__array.max(axis, keepdim=True) 
-    else:
-        return x._tensor__array.max()
 
-    return __reduce_op(x, max_axis, mpi.reduce_op.MAX, axis)
+    return __reduce_op(x, x._tensor__array.max, mpi.reduce_op.MAX, axis)
```
Comment (Markus-Goetz) [inline heat/core/operations.py]: We should use the high-level torch module function there, i.e. torch.max and then inside __reduce_op do partial_op(x). This way we can pass bound and unbound functions to __reduce_op

**Comment by ClaudiaComito** [comment]
Thanks a lot Markus, I introduced the changes you suggested, more later.

File: heat/core/operations.py (hunk 1 above)
Comment (krajsek) [reply heat/core/operations.py]: Hi Claudia, there are automatic tools taking care of PEP-8 compliance, e.g. autopep8, also some IDEs
like pycharm have these feature.

File: heat/core/operations.py [hunk 5]
```diff
@@ -393,13 +468,20 @@ def __reduce_op(x,partial, op, axis):
     # perform sanitation
     if not isinstance(x, tensor.tensor):
         raise TypeError('expected x to be a ht.tensor, but was {}'.format(type(x)))
-    
-    
+    axis = stride_tricks.sanitize_axis(x.shape, axis)
+    keepdim = True
```
[the remark this answers is not in the document]
Comment (ClaudiaComito) [reply heat/core/operations.py]: Done.

**Pull request merged**